client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx

```diff
@@ -33,15 +33,16 @@ export default React.createClass( {
 
 	render() {
 		const { site, redirect, linkDisplay } = this.props;
-		const buttonElement = linkDisplay ? 'button' : Button;
```

Comment:
I understand your concerns @aduth, but I'm actually preserving the current behavior, and just cleaning up the unnecessary logic. By directly using a `Button` we are still rendering it as a link if `linkDisplay` is `true`. I've done some manual testing on all occurrences and use cases of `DisconnectJetpackButton` to confirm there are no regressions, but this is also underlined by the following facts:

1. If `linkDisplay` was true, it was using a `<button>`, and this is what `Button` does too (since it's not called with  a `href` prop). This makes the line I've removed unnecessary.
2. I haven't touched this line: `'is-link': linkDisplay`, and this CSS class actually preserves the current behavior and look.

Considering this, I don't see a case where we would possibly have a regression. Got further thoughts, @aduth?